Automattic/wp-calypso
# Notifications: Refactor gridicons as functional component and add translations

Related to https://github.com/Automattic/wp-calypso/issues/92770

## Proposed Changes

* Refactor Gridicons as a functional component.
* Add translations.
* Update some of the titles (tooltips) to be more descriptive of what the button does, instead of what the icon is.

Example translated tooltip:

<img width="433" alt="Screen Shot 2024-12-25 at 11 00 53 AM" src="https://github.com/user-attachments/assets/3da2d8e2-f401-4859-a649-6d02d19922ea" />

Noting:
- I'm going to merge this without waiting for translations since they don't exist now, and this will at least add some existing translations.
- This will need to be deployed to Jetpack/Atomic.
- There's an issue with the undo list item close button positioning that can be addressed separately.

## Why are these changes being made?


* I noticed these aren't translated when working on https://github.com/Automattic/wp-calypso/issues/92770

## Testing Instructions



* From your WordPress.com dashboard, open the BELL notifications sidebar.
* Hover over all of the icons you see: 
    * Gear icon
    * Info icon
    * Icons on top of avatars, like reblog and comment
    * Comment approve, spam, trash, like, edit icons (seen when you click on a comment)
    * Reader following and follow (seen when you open Subscribers tab and click on a notification)
* Do the hover descriptions / tooltips make sense? Are they helpful?
* Click on the clickable icons (info, comment moderation, follow) to make sure they still function correctly.
* Test the changes on your sandbox by running `install-plugin.sh notifications update/notifications-gridicons` and sandboxing `widgets.wp.com` and `wordpress.com`. Then open /wp-admin and the Notifications panel.

## Pre-merge Checklist
